packages/react-components/src/components/HoldButton.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ControlBarButton, ControlBarButtonProps } from './ControlBarButton';
+import React from 'react';
+import { HighContrastAwareIcon } from './HighContrastAwareIcon';
+
+/* @conditional-compile-remove(PSTN-calls) */
+/**
+ *@internal
+ */
+export interface HoldButtonProps extends ControlBarButtonProps {
+  /**
+   * Utility property for using this component with communication react handlers
+   * Holds the call or resumes it based on call state.
+   */
+  onToggleHold: () => Promise<void>;
+  /**
+   * Optional strings to override in component
+   */
+  strings: HoldButtonStrings;
+}
+
+/* @conditional-compile-remove(PSTN-calls) */
+/**
+ * Strings for the hold button labels
+ * @internal
+ */
+export interface HoldButtonStrings {
+  /**
+   * Label for when action is to resume call.
+   */
+  onLabel: string;
+  /**
+   * Label for when action is to hold call.
+   */
+  offLabel: string;
+  /**
+   * Content for when button is checked
+   */
+  tooltipOnContent: string;
+  /**
+   * Content for when button is unchecked
+   */
+  toolTipOffContent: string;
+}
+
+/* @conditional-compile-remove(PSTN-calls) */
+/**
+ * a button to hold or resume a ongoing call.
+ *
+ * Can be used with {@link ControlBar}
+ *
+ * @param props - properties for the hold button.
+ * @returns
+ * @internal
+ */
+export const HoldButton = (props: HoldButtonProps): JSX.Element => {
+  const { onToggleHold, strings } = props;
+
+  const onRenderHoldIcon = (): JSX.Element => {
+    return <HighContrastAwareIcon disabled={props.disabled} iconName="HoldCall" />;
+  };
+  const onRenderResumeIcon = (): JSX.Element => {
+    return <HighContrastAwareIcon disabled={props.disabled} iconName="ResumeCall" />;
+  };
+
+  return (
+    <ControlBarButton
+      {...props}
```

Comment:
do we also want to expose any styles here as well? 

ScreenShareButton has this

![image](https://user-images.githubusercontent.com/79329532/169880252-7a688f2d-519b-4535-9a9d-81d47d4711a6.png)